### PR TITLE
Docs multiplugin page-wide tabs merge

### PR DESCRIPTION
**Pull request by kblaszczak-intel**
A new proposition for the MULTI article, aligned with the accepted AUTO version.

### CODE REVIEW

File: docs/OV_Runtime_UG/multi_device.md [hunk 1]
```diff
@@ -1,39 +1,41 @@
 # Running on multiple devices simultaneously {#openvino_docs_OV_UG_Running_on_multiple_devices}
 
-## Introducing the Multi-Device Plugin (C++)
 
-@sphinxdirective
-.. raw:: html
 
-    <div id="switcher-cpp" class="switcher-anchor">C++</div>
-@endsphinxdirective
+The Multi-Device execution mode, or MULTI for short, uses a "virtual" or a "proxy" device, which does not bind to a specific type of hardware. Instead, it assigns available computing devices to particular inference requests, which are then executed in parallel.
```
Comment (myshevts) [inline docs/OV_Runtime_UG/multi_device.md:5]: "uses" sounds a bit misleading, either "constitutes", or maybe "acts"

File: docs/OV_Runtime_UG/multi_device.md [hunk 2]
```diff
@@ -1,39 +1,41 @@
...
 
-The Multi-Device plugin automatically assigns inference requests to available computational devices to execute the requests in parallel. By contrast, the Heterogeneous plugin can run different layers on different devices but not in parallel. The potential gains with the Multi-Device plugin are:
+The potential gains from using Multi-Device execution are:
+* improved throughput from using multiple devices at once,
+* increase in performance stability due to multiple devices sharing inference workload.
 
-* Improved throughput from using multiple devices (compared to single-device execution)
-* More consistent performance, since the devices share the inference burden (if one device is too busy, another can take more of the load)
+What is important, the Multi-Device mode does not change tha application logic, so it does not require you to explicitly compile the model on every device or create and balance inference requests. From the application point of view, this is just another plugin that handles the actual machinery.
```
Comment (myshevts) [inline docs/OV_Runtime_UG/multi_device.md:11]: "just another plugin that handles " - > "just a conventional **device** that **internally** handles the actual machinery"

File: docs/OV_Runtime_UG/multi_device.md [hunk 3]
```diff
@@ -1,39 +1,41 @@
...
+Following the OpenVINO™ naming convention, the Multi-Device mode is assigned the label of “MULTI.” The only configuration option available for it is a prioritized list of devices to use:
 
-### Defining and Configuring the Multi-Device Plugin
+@sphinxdirective
 
-Following the OpenVINO™ convention of labeling devices, the Multi-Device plugin uses the name "MULTI". The only configuration option for the Multi-Device plugin is a prioritized list of devices to use:
++---------------------------+---------------------------------+-----------------------------------------------------------+
+| Property                  | Property values                 | Description                                               |
++===========================+=================================+===========================================================+
+| <device candidate list>   | | AUTO: <device names>          || Specifies the devices available for selection.           |
```
Comment (myshevts) [inline docs/OV_Runtime_UG/multi_device.md:25]: MULTI:

File: docs/OV_Runtime_UG/multi_device.md [hunk 4]
```diff
@@ -1,39 +1,41 @@
...
+|                           | | comma-separated, no spaces    |                                                           |
++---------------------------+---------------------------------+-----------------------------------------------------------+
 
-| Parameter name | Parameter values | Default | Description |
-| -------------- | ---------------- | --- | --- |
-| ov::device::priorities | comma-separated device names with no spaces | N/A | Prioritized list of devices |
+@endsphinxdirective
 
-You can set the priorities directly as a string.
+The device candidate list allows users to customize the priority and limit the choice of devices available to MULTI. If not specified, the plugin assumes all the devices present in the system can be used. Importantly, the list may also specify the number of requests for MULTI to keep for each device, as described below.
```
Comment (myshevts) [inline docs/OV_Runtime_UG/multi_device.md:34]: " If not specified, the plugin assumes all the devices present in the system can be used"- this is not true, unlike the AUTO, the MULTI does require the explicit list of the devices

File: docs/OV_Runtime_UG/multi_device.md (hunk 3 above)
Comment (myshevts) [inline docs/OV_Runtime_UG/multi_device.md:25]: unlike AUTO, these are NOT candidates: if loading to any device from the list fails, the entire MULTI would fail as well

File: docs/OV_Runtime_UG/multi_device.md (hunk 4 above)
Comment (myshevts) [reply docs/OV_Runtime_UG/multi_device.md:34]: pls see above, the list is not "candidates".

File: docs/OV_Runtime_UG/multi_device.md [hunk 5]
```diff
@@ -137,186 +115,46 @@ You can use the [configure devices](supported_plugins/config_properties.md) to q
...
-
-
-### Enumerating Available Devices
-The OpenVINO Runtime API features a dedicated methods to enumerate devices and their capabilities. See the [Hello Query Device Python Sample](../../samples/python/hello_query_device/README.md). This is example output from the sample (truncated to device names only):
+To see how the Multi-Device execution is used in practice and test its performance, take a look at OpenVINO™ samples. All samples supporting the `-d` command-line option (which stands for “device”) will accept the plugin out-of-the-box. The Benchmark Application will be a perfect place to start – it presents the optimal performance of the plugin without the need for additional settings, like the number of requests or CPU threads.
```
Comment (myshevts) [inline docs/OV_Runtime_UG/multi_device.md:122]: "will accept the plugin out-of-the-box" is confusing... we are usually talking about the "devices", rather than "plugins".
the benchmark_app is today the on;ly option, as the rest of the samples are API examples, and the MULTI is about  the performance, whihc only the benchamrk_app reports.

File: docs/OV_Runtime_UG/multi_device.md (hunk 1 above)
Comment (kblaszczak-intel) [inline docs/OV_Runtime_UG/multi_device.md:5]:
```suggestion
The Multi-Device execution mode, or MULTI for short, acts as a "virtual" or a "proxy" device, which does not bind to a specific type of hardware. Instead, it assigns available computing devices to particular inference requests, which are then executed in parallel. 
```

File: docs/OV_Runtime_UG/multi_device.md [hunk 6]
```diff
@@ -137,186 +115,46 @@ You can use the [configure devices](supported_plugins/config_properties.md) to q
...
 
 @sphinxdirective
+.. note::
 
-.. tab:: Python
-
-    .. doxygensnippet:: docs/snippets/ov_multi.py
-       :language: python
-       :fragment: [available_devices_2]
+   You can use the FP16 IR to work with auto-device.
```
Comment (myshevts) [inline docs/OV_Runtime_UG/multi_device.md:134]: MULTI

File: docs/OV_Runtime_UG/multi_device.md [hunk 7]
```diff
@@ -1,39 +1,41 @@
...
 
-The Multi-Device plugin automatically assigns inference requests to available computational devices to execute the requests in parallel. By contrast, the Heterogeneous plugin can run different layers on different devices but not in parallel. The potential gains with the Multi-Device plugin are:
+The potential gains from using Multi-Device execution are:
+* improved throughput from using multiple devices at once,
+* increase in performance stability due to multiple devices sharing inference workload.
 
-* Improved throughput from using multiple devices (compared to single-device execution)
-* More consistent performance, since the devices share the inference burden (if one device is too busy, another can take more of the load)
+What is important, the Multi-Device mode does not change tha application logic, so it does not require you to explicitly compile the model on every device or create and balance inference requests. From the application point of view, this is just another plugin that handles the actual machinery.
```
Comment (kblaszczak-intel) [inline docs/OV_Runtime_UG/multi_device.md:11]:
```suggestion
What is important, the Multi-Device mode does not change the application logic, so it does not require you to explicitly compile the model on every device or create and balance inference requests. From the application point of view, this is just a conventional device that internally handles the actual machinery.
```

File: docs/OV_Runtime_UG/multi_device.md (hunk 6 above)
Comment (myshevts) [reply docs/OV_Runtime_UG/multi_device.md:134]: the message is that no need to use the FP32 IR when any device in the list doesn't suppurt the FP16, the conversion is automacally happens by the OV

File: docs/OV_Runtime_UG/multi_device.md [hunk 8]
```diff
@@ -1,39 +1,41 @@
...
+Following the OpenVINO™ naming convention, the Multi-Device mode is assigned the label of “MULTI.” The only configuration option available for it is a prioritized list of devices to use:
 
-### Defining and Configuring the Multi-Device Plugin
+@sphinxdirective
 
-Following the OpenVINO™ convention of labeling devices, the Multi-Device plugin uses the name "MULTI". The only configuration option for the Multi-Device plugin is a prioritized list of devices to use:
++---------------------------+---------------------------------+-----------------------------------------------------------+
+| Property                  | Property values                 | Description                                               |
++===========================+=================================+===========================================================+
+| <device candidate list>   | | AUTO: <device names>          || Specifies the devices available for selection.           |
```
Comment (kblaszczak-intel) [inline docs/OV_Runtime_UG/multi_device.md:25]:
```suggestion
| <device candidate list>   | | MULTI: <device names>         || Specifies the devices available for selection.           |
```

**Comment by myshevts** [comment]
@wangleis we discussed that there is a need to add a note on the performance hints (that the MULTI propagates to every device in the list)

File: docs/OV_Runtime_UG/multi_device.md (hunk 4 above)
Comment (kblaszczak-intel) [reply docs/OV_Runtime_UG/multi_device.md:34]: Oh, ok, that is an important difference. Repairing it now.

File: docs/OV_Runtime_UG/multi_device.md [hunk 9]
```diff
@@ -1,39 +1,41 @@
...
+Following the OpenVINO™ naming convention, the Multi-Device mode is assigned the label of “MULTI.” The only configuration option available for it is a prioritized list of devices to use:
 
-### Defining and Configuring the Multi-Device Plugin
+@sphinxdirective
 
-Following the OpenVINO™ convention of labeling devices, the Multi-Device plugin uses the name "MULTI". The only configuration option for the Multi-Device plugin is a prioritized list of devices to use:
++---------------------------+---------------------------------+-----------------------------------------------------------+
+| Property                  | Property values                 | Description                                               |
++===========================+=================================+===========================================================+
+| <device candidate list>   | | MULTI: <device names>         || Specifies the devices available for selection.           |
```
Comment (kblaszczak-intel) [inline docs/OV_Runtime_UG/multi_device.md:25]:
```suggestion
| <device list>             | | MULTI: <device names>         || Specifies the devices available for selection.           |
```

File: docs/OV_Runtime_UG/multi_device.md [hunk 10]
```diff
@@ -1,39 +1,41 @@
...
+|                           | | comma-separated, no spaces    |                                                           |
++---------------------------+---------------------------------+-----------------------------------------------------------+
 
-| Parameter name | Parameter values | Default | Description |
-| -------------- | ---------------- | --- | --- |
-| ov::device::priorities | comma-separated device names with no spaces | N/A | Prioritized list of devices |
+@endsphinxdirective
 
-You can set the priorities directly as a string.
+The device candidate list allows users to customize the priority and limit the choice of devices available to MULTI. If not specified, the plugin assumes all the devices present in the system can be used. Importantly, the list may also specify the number of requests for MULTI to keep for each device, as described below.
```
Comment (kblaszczak-intel) [inline docs/OV_Runtime_UG/multi_device.md:34]:
```suggestion
Specifying the device list explicitly is required by MULTI, as it defines the devices available for inference and sets their priorities.  Importantly, the list may also specify the number of requests for MULTI to keep for each device, as described below.
```

File: docs/OV_Runtime_UG/multi_device.md (hunk 5 above)
Comment (kblaszczak-intel) [inline docs/OV_Runtime_UG/multi_device.md:122]:
```suggestion
To see how the Multi-Device execution is used in practice and test its performance, take a look at OpenVINO™ samples. All samples supporting the `-d` command-line option (which stands for “device”) will accept the MULTI mode natively. The Benchmark Application will be a perfect place to start – it presents the optimal performance of the plugin without the need for additional settings, like the number of requests or CPU threads. 
```

Comment (kblaszczak-intel) [reply docs/OV_Runtime_UG/multi_device.md:122]: I guess "will accept the MULTI mode natively" is a better choice of words. Is it ok? 
But does it mean we should not mention the '-d" option? If so, we can remove the whole sentence.

File: docs/OV_Runtime_UG/multi_device.md (hunk 1 above)
Comment (kblaszczak-intel) [reply docs/OV_Runtime_UG/multi_device.md]: nice change, done.

File: docs/OV_Runtime_UG/multi_device.md (hunk 2 above)
Comment (kblaszczak-intel) [reply docs/OV_Runtime_UG/multi_device.md]: sounds better and more formal, gladly changing... done.

File: docs/OV_Runtime_UG/multi_device.md (hunk 3 above)
Comment (kblaszczak-intel) [reply docs/OV_Runtime_UG/multi_device.md]: done

File: docs/OV_Runtime_UG/multi_device.md [hunk 11]
```diff
@@ -137,186 +115,46 @@ You can use the [configure devices](supported_plugins/config_properties.md) to q
...
 
 @sphinxdirective
+.. note::
 
-.. tab:: Python
-
-    .. doxygensnippet:: docs/snippets/ov_multi.py
-       :language: python
-       :fragment: [available_devices_2]
+   You can use the FP16 IR to work with auto-device.
```
Comment (kblaszczak-intel) [inline docs/OV_Runtime_UG/multi_device.md:134]:
```suggestion
   You can keep using the FP16 IR, without converting it to FP32, to work with MULTI, even if some of the listed devices do not support it. The conversion will be done automatically for you.
```

File: docs/OV_Runtime_UG/multi_device.md (hunk 6 above)
Comment (kblaszczak-intel) [reply docs/OV_Runtime_UG/multi_device.md:134]: The whole point reworded, I hope now it's more clear

File: docs/OV_Runtime_UG/multi_device.md [hunk 12]
```diff
@@ -137,186 +115,46 @@ You can use the [configure devices](supported_plugins/config_properties.md) to q
...
-
-
-### Enumerating Available Devices
-The OpenVINO Runtime API features a dedicated methods to enumerate devices and their capabilities. See the [Hello Query Device Python Sample](../../samples/python/hello_query_device/README.md). This is example output from the sample (truncated to device names only):
+To see how the Multi-Device execution is used in practice and test its performance, take a look at OpenVINO™ samples. All samples supporting the `-d` command-line option (which stands for “device”) will accept the MULTI mode natively. The Benchmark Application will be a perfect place to start – it presents the optimal performance of the plugin without the need for additional settings, like the number of requests or CPU threads.
```
Comment (kblaszczak-intel) [inline docs/OV_Runtime_UG/multi_device.md:122]:
```suggestion
To see how the Multi-Device execution is used in practice and test its performance, take a look at OpenVINO's Benchmark Application which presents the optimal performance of the plugin without the need for additional settings, like the number of requests or CPU threads. 
```

File: docs/OV_Runtime_UG/supported_plugins/Device_Plugins.md [hunk 13]
```diff
@@ -54,3 +54,52 @@ The table below demonstrates support of key features by OpenVINO device plugins.
 | [Extensibility](@ref openvino_docs_Extensibility_UG_Intro) | Yes | Yes | No | No |
 
 For more details on plugin specific feature limitation, see corresponding plugin pages.
+
+
+
+## Enumerating Available Devices
+The OpenVINO Runtime API features dedicated methods of enumerating devices and their capabilities. See the [Hello Query Device C++ Sample](../../samples/cpp/hello_query_device/README.md). This is example output from the sample (truncated to device names only):
```
Comment (kblaszczak-intel) [inline docs/OV_Runtime_UG/supported_plugins/Device_Plugins.md:61]:
```suggestion
The OpenVINO Runtime API features dedicated methods of enumerating devices and their capabilities. See the [Hello Query Device C++ Sample](../../../samples/cpp/hello_query_device/README.md). This is example output from the sample (truncated to device names only):
```

File: docs/snippets/MULTI0.cpp [hunk 14]
```diff
@@ -3,17 +3,23 @@
 int main() {
 //! [part0]
 ov::Core core;
+
+// Read a network in IR, PaddlePaddle, or ONNX format:
 std::shared_ptr<ov::Model> model = core.read_model("sample.xml");
-// the "MULTI" device is (globally) pre-configured with the explicit option
-core.set_property("MULTI", ov::device::priorities("HDDL,GPU"));
```
Comment (songbell) [inline docs/snippets/MULTI0.cpp:8]: if delete these 2 lines, below does not work
ov::CompiledModel compileModel0 = core.compile_model(model, "MULTI");
1) we can compile with MULTI + device priority config
2) we can pre-set the config by core.set_property("MULTI", ov::device::priorities(*****));
    then, we can compile Model to MULTI as below:
    ov::CompiledModel compileModel0 = core.compile_model(model, "MULTI");

File: docs/snippets/MULTI1.cpp [hunk 15]
```diff
@@ -5,18 +5,25 @@ int main() {
...
 // reverse the order of priorities
 compileModel.set_property(ov::device::priorities("GPU,HDDL"));
-// you can even exclude some device (HDDL here)
+
+// exclude some devices (in this case, HDDL)
 compileModel.set_property(ov::device::priorities("GPU"));
-// and then return it back
+
+// return excluded devices
 compileModel.set_property(ov::device::priorities("GPU,HDDL"));
```
Comment (songbell) [inline docs/snippets/MULTI1.cpp:17]: suggest change to :
// include back the excluded devices

File: docs/snippets/MULTI4.cpp [hunk 16]
```diff
@@ -3,18 +3,19 @@
...
-ov::CompiledModel compileModel = core.compile_model(model, "MULTI",
+
+// When compiling the model on MULTI, configure CPU and MYRIAD 
+// (devices, priorities, and device configurations):
+ov::CompiledModel compiled_model = core.compile_model(model, "MULTI",
     ov::device::priorities("HDDL", "GPU"),
-    ov::device::properties("HDDL", hddl_config),
-    ov::device::properties("GPU", gpu_config));
+    ov::device::properties("CPU", cpu_config),
+    ov::device::properties("MYRIAD", myriad_config));
```
Comment (songbell) [inline docs/snippets/MULTI4.cpp:16]: these codes are illustrating that when we load to multi devices like HDDL and GPU, we can specify extra configs for each target device as well, like HDDL with hddl_config and GPU with gpu_config, multi plugin will then load the network to each target device with these pre-set configs. 
It's not about configure CPU or MYRIAD in this case

File: docs/snippets/ov_multi.py [hunk 17]
```diff
@@ -2,28 +2,53 @@
...
+def MULTI_0():
+#! [MULTI_0]
     core = Core()
 
-    # Read a network in IR or ONNX format
+    # Read a network in IR, PaddlePaddle, or ONNX format:
     model = core.read_model(model_path)
+    
+    # compile a model on MULTI using the default list of device candidates.
+    compiled_model = core.compile_model(model=model, device_name="MULTI")
```
Comment (songbell) [inline docs/snippets/ov_multi.py:13]: same as above, please see comment in cpp code snipet. either has to set the device from compile_model or from core.set_property

File: docs/snippets/ov_multi.py [hunk 18]
```diff
@@ -2,28 +2,53 @@
...
-#! [Option_1]
+    compiled_model = core.compile_model(model=model, device_name="MULTI", config={"MULTI_DEVICE_PRIORITIES": "GPU,CPU"})
 
-def Option_2():
-#! [Option_2]
-    core = Core()
+    # Optional
+    # MULTI is pre-configured (globally) with the explicit option.
+    
+    core.set_property(device_name="AUTO", properties={"MULTI_DEVICE_PRIORITIES":"GPU,CPU"})
```
Comment (songbell) [inline docs/snippets/ov_multi.py:24]: this is multi code snippet, no "AUTO" should be involved

File: docs/snippets/MULTI0.cpp (hunk 14 above)
Comment (kblaszczak-intel) [reply docs/snippets/MULTI0.cpp:8]: Done. Additional comments changes should make things even clearer now. Thanks!

File: docs/snippets/MULTI1.cpp (hunk 15 above)
Comment (kblaszczak-intel) [reply docs/snippets/MULTI1.cpp:17]: done with minor rewording :)

File: docs/snippets/MULTI4.cpp (hunk 16 above)
Comment (kblaszczak-intel) [reply docs/snippets/MULTI4.cpp:16]: Now when I'm looking at it, I have no bloody idea why it ended up like that! I have aligned the c++ and Python versions.

File: docs/snippets/ov_multi.py (hunk 17 above)
Comment (kblaszczak-intel) [reply docs/snippets/ov_multi.py:13]: done

File: docs/snippets/ov_multi.py (hunk 18 above)
Comment (kblaszczak-intel) [reply docs/snippets/ov_multi.py:24]: well spotted, thanks!

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 19]
```diff
@@ -66,12 +66,12 @@ benchmark_app -m ../public/alexnet/FP32/alexnet.xml -d AUTO -niter 128
 
 @sphinxdirective
 .. note::
+
    The longer the process runs, the closer realtime performance will be to that of the best-suited device.
 @endsphinxdirective
 
 ## Using the Auto-Device Plugin
```
Comment (xu-yuan1) [inline docs/OV_Runtime_UG/auto_device_selection.md:73]: Do we need change to Mode here?

File: docs/OV_Runtime_UG/multi_device.md [hunk 20]
```diff
@@ -1,39 +1,41 @@
...
-| Parameter name | Parameter values | Default | Description |
-| -------------- | ---------------- | --- | --- |
-| ov::device::priorities | comma-separated device names with no spaces | N/A | Prioritized list of devices |
+@endsphinxdirective
 
-You can set the priorities directly as a string.
+Specifying the device list explicitly is required by MULTI, as it defines the devices available for inference and sets their priorities.  Importantly, the list may also specify the number of requests for MULTI to keep for each device, as described below.
 
-Basically, there are three ways to specify the devices to be use by the "MULTI":
+Note, that OpenVINO™ Runtime lets you use “GPU” as an alias for “GPU.0” in function calls. More detail on enumerating devices can be found in [Working with devices](supported_plugins/Device_Plugins.md).
```
Comment (xu-yuan1) [inline docs/OV_Runtime_UG/multi_device.md:36]:
```suggestion
Note that OpenVINO™ Runtime lets you use “GPU” as an alias for “GPU.0” in function calls. More details on enumerating devices can be found in [Working with devices](supported_plugins/Device_Plugins.md).
```

File: docs/OV_Runtime_UG/multi_device.md [hunk 21]
```diff
@@ -43,9 +45,15 @@ Basically, there are three ways to specify the devices to be use by the "MULTI":
...
+.. tab:: Python
+
+    .. doxygensnippet:: docs/snippets/ov_multi.py
+       :language: python
+       :fragment: [MULTI_0]
+
 @endsphinxdirective
 
-Notice that the priorities of the devices can be changed in real time for the compiled model:
+Notice that MULTI allows you to **change device priorities on the fly**. You can alter the order, exclude a device, and bring an excluded device back. Still, it does not allow to add new devices.
```
Comment (xu-yuan1) [inline docs/OV_Runtime_UG/multi_device.md:56]:
```suggestion
Notice that MULTI allows you to **change device priorities on the fly**. You can alter the order, exclude a device, and bring an excluded device back. Still, it does not allow adding new devices.
```

File: docs/OV_Runtime_UG/multi_device.md [hunk 22]
```diff
@@ -1,39 +1,41 @@
...
 
-The Multi-Device plugin automatically assigns inference requests to available computational devices to execute the requests in parallel. By contrast, the Heterogeneous plugin can run different layers on different devices but not in parallel. The potential gains with the Multi-Device plugin are:
+The potential gains from using Multi-Device execution are:
+* improved throughput from using multiple devices at once,
+* increase in performance stability due to multiple devices sharing inference workload.
 
-* Improved throughput from using multiple devices (compared to single-device execution)
-* More consistent performance, since the devices share the inference burden (if one device is too busy, another can take more of the load)
+What is important, the Multi-Device mode does not change the application logic, so it does not require you to explicitly compile the model on every device or create and balance inference requests. From the application point of view, this is just a conventional device that internally handles the actual machinery.
```
Comment (xu-yuan1) [inline docs/OV_Runtime_UG/multi_device.md:11]:
```suggestion
Importantly, the Multi-Device mode does not change the application logic, so it does not require you to explicitly compile the model on every device or create and balance inference requests. To the application, this is just a conventional device that internally handles the actual devices.
```

File: docs/OV_Runtime_UG/multi_device.md [hunk 23]
```diff
@@ -1,39 +1,41 @@
...
 
-Note that with Multi-Device the application logic is left unchanged, so you don't need to explicitly compile the model on every device, create and balance the inference requests and so on. From the application point of view, this is just another device that handles the actual machinery. The only thing that is required to leverage performance is to provide the multi-device (and hence the underlying devices) with enough inference requests to process. For example, if you were processing 4 cameras on the CPU (with 4 inference requests), it might be desirable to process more cameras (with more requests in flight) to keep CPU and GPU busy via Multi-Device.
+Note that performance increase in this mode comes from utilizing multiple devices at once. This means that you need to provide the devices with enough inference requests to keep them busy or you will not benefit much from using MULTI.
```
Comment (xu-yuan1) [inline docs/OV_Runtime_UG/multi_device.md:13]:
```suggestion
Note that the performance increase in this mode comes from utilizing multiple devices at once. This means that you need to provide the devices with enough inference requests to keep them busy, otherwise you will not benefit much from using MULTI.
```

File: docs/OV_Runtime_UG/multi_device.md [hunk 24]
```diff
@@ -55,59 +63,23 @@ Notice that the priorities of the devices can be changed in real time for the co
...
-...
-    Device: MYRIAD.1.4-ma2480
-```
 
-So the explicit configuration to use both would be "MULTI:MYRIAD.1.2-ma2480,MYRIAD.1.4-ma2480". Accordingly, the code that loops over all available devices of "MYRIAD" type only is below:
-
-@sphinxdirective
 
-.. tab:: C++
+One more thing you can define is the **number of requests to allocate for each device**. You can do it simply by adding the number to each device in parentheses, like so: `"MULTI:CPU(2),GPU(2)"`. Although, this method is not recommended as it is not performance-portable. The suggested approach is to configure individual devices and query the resulting number of requests to be used at the application level, as described below, in [Configuring Individual Devices and Creating MULTI On Top](#configuring-the-individual-devices-and-creating-the-multi-device-on-top).
```
Comment (xu-yuan1) [inline docs/OV_Runtime_UG/multi_device.md:76]:
```suggestion
One more thing you can define is the **number of requests to allocate for each device**. You can do it simply by adding the number to each device in parentheses, like this: `"MULTI:CPU(2),GPU(2)"`. However, this method is not recommended as it is not performance-portable. The suggested approach is to configure individual devices and query the resulting number of requests to be used at the application level, as described in [Configuring Individual Devices and Creating MULTI On Top](#configuring-the-individual-devices-and-creating-the-multi-device-on-top).
```

File: docs/OV_Runtime_UG/multi_device.md [hunk 25]
```diff
@@ -55,59 +63,23 @@ Notice that the priorities of the devices can be changed in real time for the co
...
 
-    .. doxygensnippet:: docs/snippets/MULTI3.cpp
-       :language: cpp
-       :fragment: [part3]
+To check what devices are present in the system, you can use the Device API. For information on how to do it, check [Query device properties and configuration](supported_plugins/config_properties.md)
```
Comment (xu-yuan1) [inline docs/OV_Runtime_UG/multi_device.md:78]:
```suggestion
To check what devices are present in the system, you can use the Device API. For information on how to do it, check [Query device properties and configuration](supported_plugins/config_properties.md).
```

File: docs/OV_Runtime_UG/supported_plugins/Device_Plugins.md [hunk 26]
```diff
@@ -53,4 +53,53 @@ The table below demonstrates support of key features by OpenVINO device plugins.
 | [Stateful models](../network_state_intro.md) | Yes | No | Yes | No |
 | [Extensibility](@ref openvino_docs_Extensibility_UG_Intro) | Yes | Yes | No | No |
 
-For more details on plugin-specific feature limitations, see the corresponding plugin pages.
+For more details on plugin specific feature limitation, see corresponding plugin pages.
```
Comment (xu-yuan1) [inline docs/OV_Runtime_UG/supported_plugins/Device_Plugins.md:56]: ### 
```suggestion
For more details on plugin-specific feature limitations, see corresponding plugin pages.
```

File: docs/OV_Runtime_UG/supported_plugins/Device_Plugins.md [hunk 27]
```diff
@@ -53,4 +53,53 @@ The table below demonstrates support of key features by OpenVINO device plugins.
...
+
+Beyond the trivial "CPU", "GPU", "HDDL" and so on, when multiple instances of a device are available the names are more qualified. For example, this is how two Intel® Movidius™ Myriad™ X sticks are listed with the hello_query_sample:
+```
+...
+    Device: MYRIAD.1.2-ma2480
+...
+    Device: MYRIAD.1.4-ma2480
+```
+
+So the explicit configuration to use both would be "MULTI:MYRIAD.1.2-ma2480,MYRIAD.1.4-ma2480". Accordingly, the code that loops over all available devices of "MYRIAD" type only is below:
```
Comment (xu-yuan1) [inline docs/OV_Runtime_UG/supported_plugins/Device_Plugins.md:95]: ### 
```suggestion
So the explicit configuration to use both would be "MULTI:MYRIAD.1.2-ma2480,MYRIAD.1.4-ma2480". Accordingly, the code that loops over all available devices of "MYRIAD" type only is as follows:
```

File: docs/OV_Runtime_UG/supported_plugins/Device_Plugins.md [hunk 28]
```diff
@@ -53,4 +53,53 @@ The table below demonstrates support of key features by OpenVINO device plugins.
 | [Stateful models](../network_state_intro.md) | Yes | No | Yes | No |
 | [Extensibility](@ref openvino_docs_Extensibility_UG_Intro) | Yes | Yes | No | No |
 
-For more details on plugin-specific feature limitations, see the corresponding plugin pages.
+For more details on plugin specific feature limitation, see corresponding plugin pages.
+
+
+
+## Enumerating Available Devices
+The OpenVINO Runtime API features dedicated methods of enumerating devices and their capabilities. See the [Hello Query Device C++ Sample](../../../samples/cpp/hello_query_device/README.md). This is example output from the sample (truncated to device names only):
```
Comment (xu-yuan1) [inline docs/OV_Runtime_UG/supported_plugins/Device_Plugins.md:61]:
```suggestion
The OpenVINO Runtime API features dedicated methods of enumerating devices and their capabilities. See the [Hello Query Device C++ Sample](../../../samples/cpp/hello_query_device/README.md). This is the example output from the sample (truncated to device names only):
```

File: docs/OV_Runtime_UG/supported_plugins/Device_Plugins.md [hunk 29]
```diff
@@ -53,4 +53,53 @@ The table below demonstrates support of key features by OpenVINO device plugins.
...
+
+.. tab:: C++
+
+    .. doxygensnippet:: docs/snippets/MULTI2.cpp
+       :language: cpp
+       :fragment: [part2]
+
+@endsphinxdirective
+
+Beyond the trivial "CPU", "GPU", "HDDL" and so on, when multiple instances of a device are available the names are more qualified. For example, this is how two Intel® Movidius™ Myriad™ X sticks are listed with the hello_query_sample:
```
Comment (xu-yuan1) [inline docs/OV_Runtime_UG/supported_plugins/Device_Plugins.md:87]:
```suggestion
Beyond the trivial "CPU", "GPU", "HDDL" and so on, when multiple instances of a device are available, the names are more qualified. For example, this is how two Intel® Movidius™ Myriad™ X sticks are listed with the hello_query_sample:
```

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 19 above)
Comment (kblaszczak-intel) [inline docs/OV_Runtime_UG/auto_device_selection.md:73]:
```suggestion
## Using the Auto-Device Mode 
```

File: docs/OV_Runtime_UG/multi_device.md (hunk 22 above)
Comment (kblaszczak-intel) [reply docs/OV_Runtime_UG/multi_device.md:11]:
```suggestion
Importantly, the Multi-Device mode does not change the application logic, so it does not require you to explicitly compile the model on every device or create and balance inference requests. It appears to use a typical device but internally handles the actual hardware.
```

File: docs/OV_Runtime_UG/supported_plugins/Device_Plugins.md (hunk 28 above)
Comment (kblaszczak-intel) [reply docs/OV_Runtime_UG/supported_plugins/Device_Plugins.md:61]:
```suggestion
The OpenVINO Runtime API features dedicated methods of enumerating devices and their capabilities. See the [Hello Query Device C++ Sample](../../../samples/cpp/hello_query_device/README.md). This is an example output from the sample (truncated to device names only):
```

File: docs/OV_Runtime_UG/supported_plugins/Device_Plugins.md (hunk 27 above)
Comment (kblaszczak-intel) [reply docs/OV_Runtime_UG/supported_plugins/Device_Plugins.md:95]:
```suggestion
So, the explicit configuration to use both would be "MULTI:MYRIAD.1.2-ma2480,MYRIAD.1.4-ma2480". Accordingly, the code that loops over all available devices of the "MYRIAD" type only is as follows:
```

File: docs/OV_Runtime_UG/supported_plugins/Device_Plugins.md (hunk 29 above)
Comment (kblaszczak-intel) [reply docs/OV_Runtime_UG/supported_plugins/Device_Plugins.md:87]:
```suggestion
Beyond the typical "CPU", "GPU", "HDDL", and so on, when multiple instances of a device are available, the names are more qualified. For example, this is how two Intel® Movidius™ Myriad™ X sticks are listed with the hello_query_sample:
```

**Comment by ilya-lavrenov** [comment]
@kblaszczak-intel please, add label that the PR is ported to master :)
